src/backend/opencl/kernel/iops.cl

```diff
@@ -46,8 +56,9 @@ void binOp(T *lhs, uint *lidx, T rhs, uint ridx)
 #ifdef MAX_OP
 void binOp(T *lhs, uint *lidx, T rhs, uint ridx)
 {
-    if ((*lhs < rhs) ||
-        (*lhs == rhs && *lidx > ridx)) {
+    if (((*lhs < rhs) ||
+         (*lhs == rhs && *lidx > ridx)) &&
+        !is_nan(*lhs)) {
```

Comment:
Is rhs guaranteed to be a non-nan number? Do we need to add a check for that as well?